### ISSUE TITLE
执行bash run_scripts/muge_finetune_vit-b-16_rbt-base.sh /home/liujiawei/datapath报错

### ISSUE THREAD

**Issue by lll1024** [OFA-Sys/Chinese-CLIP#315]
usage: run.py [-h] [--nnodes NNODES] [--nproc-per-node NPROC_PER_NODE] [--rdzv-backend RDZV_BACKEND]
              [--rdzv-endpoint RDZV_ENDPOINT] [--rdzv-id RDZV_ID] [--rdzv-conf RDZV_CONF] [--standalone]
              [--max-restarts MAX_RESTARTS] [--monitor-interval MONITOR_INTERVAL] [--start-method {spawn,fork,forkserver}]
              [--role ROLE] [-m] [--no-python] [--run-path] [--log-dir LOG_DIR] [-r REDIRECTS] [-t TEE]
              [--local-ranks-filter LOCAL_RANKS_FILTER] [--node-rank NODE_RANK] [--master-addr MASTER_ADDR]
              [--master-port MASTER_PORT] [--local-addr LOCAL_ADDR] [--logs-specs LOGS_SPECS]
              training_script ...
run.py: error: ambiguous option: --logs=/home/liujiawei/datapath/experiments/ could match --logs-specs, --logs_specs
求指导

**Comment by fangxin2github** [comment]
你好，这个问题您是怎么解决的？

**Comment by MickeyFei** [comment]
change torch version ，don t use 2.3.0

**Comment by lll1024** [comment]
> 你好，这个问题您是怎么解决的？
把--logs-specs删掉就好了

**Comment by colorfulandcjy0806** [comment]
还有一种解决方案：
将
`python3 -m torch.distributed.launch --use_env --nproc_per_node=${GPUS_PER_NODE} --nnodes=${WORKER_CNT} --node_rank=${RANK} \
          --master_addr=${MASTER_ADDR} --master_port=${MASTER_PORT} cn_clip/training/main.py \
          --train-data=${train_data} \
          --val-data=${val_data} \
          --resume=${resume} \
          ${reset_data_offset} \
          ${reset_optimizer} \
          --logs-specs=${output_base_dir} \
          --name=${name} \
          --save-step-frequency=${save_step_frequency} \
          --save-epoch-frequency=${save_epoch_frequency} \
          --log=${log_interval} \
          ${report_training_batch_acc} \
          --context-length=${context_length} \
          --warmup=${warmup} \
          --batch-size=${batch_size} \
          --valid-batch-size=${valid_batch_size} \
          --valid-step-interval=${valid_step_interval} \
          --valid-epoch-interval=${valid_epoch_interval} \
          --accum-freq=${accum_freq} \
          --lr=${lr} \
          --wd=${wd} \
          --max-epochs=${max_epochs} \
          --vision-model=${vision_model} \
          ${use_augment} \
          --text-model=${text_model}`
中的
`--log=${log_interval} `
修改为
`--logs-specs=${output_base_dir} `
这样也是可以的~

**Comment by fanshuaiyao** [comment]
> > 你好，这个问题您是怎么解决的？
> > 把--logs-specs删掉就好了

--logs-specs 请问这个参数在哪里  我咋没找到

**Comment by fanshuaiyao** [comment]
> 还有一种解决方案： 将 `python3 -m torch.distributed.launch --use_env --nproc_per_node=${GPUS_PER_NODE} --nnodes=${WORKER_CNT} --node_rank=${RANK} \ --master_addr=${MASTER_ADDR} --master_port=${MASTER_PORT} cn_clip/training/main.py \ --train-data=${train_data} \ --val-data=${val_data} \ --resume=${resume} \ ${reset_data_offset} \ ${reset_optimizer} \ --logs-specs=${output_base_dir} \ --name=${name} \ --save-step-frequency=${save_step_frequency} \ --save-epoch-frequency=${save_epoch_frequency} \ --log=${log_interval} \ ${report_training_batch_acc} \ --context-length=${context_length} \ --warmup=${warmup} \ --batch-size=${batch_size} \ --valid-batch-size=${valid_batch_size} \ --valid-step-interval=${valid_step_interval} \ --valid-epoch-interval=${valid_epoch_interval} \ --accum-freq=${accum_freq} \ --lr=${lr} \ --wd=${wd} \ --max-epochs=${max_epochs} \ --vision-model=${vision_model} \ ${use_augment} \ --text-model=${text_model}` 中的 `--log=${log_interval} `直接注释掉

不行啊   注释掉后报错：(clip-cn) fanzhengjie@zkyd:~/fsy/Chinese-CLIP$ bash run_scripts/muge_finetune_vit-b-16_rbt-base.sh /home/fanzhengjie/fsy/Clip-Data
/home/fanzhengjie/.conda/envs/clip-cn/lib/python3.10/site-packages/torch/distributed/launch.py:208: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use-env is set by default in torchrun.
If your script expects `--local-rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See
https://pytorch.org/docs/stable/distributed.html#launch-utility for
further instructions

  main()
usage: launch.py [-h] [--nnodes NNODES] [--nproc-per-node NPROC_PER_NODE] [--rdzv-backend RDZV_BACKEND] [--rdzv-endpoint RDZV_ENDPOINT] [--rdzv-id RDZV_ID]
                 [--rdzv-conf RDZV_CONF] [--standalone] [--max-restarts MAX_RESTARTS] [--monitor-interval MONITOR_INTERVAL]
                 [--start-method {spawn,fork,forkserver}] [--role ROLE] [-m] [--no-python] [--run-path] [--log-dir LOG_DIR] [-r REDIRECTS] [-t TEE]
                 [--local-ranks-filter LOCAL_RANKS_FILTER] [--node-rank NODE_RANK] [--master-addr MASTER_ADDR] [--master-port MASTER_PORT]
                 [--local-addr LOCAL_ADDR] [--logs-specs LOGS_SPECS] [--use-env]
                 training_script ...
launch.py: error: ambiguous option: --logs=/home/fanzhengjie/fsy/Clip-Data/experiments/ could match --logs-specs, --logs_specs

**Comment by MickeyFei** [comment]
把torch版本降低，2.1.0就可以了

发自我的 iPhone

在 2024年8月29日，16:37，shuaiyaofan ***@***.***> 写道：

﻿

还有一种解决方案： 将 python3 -m torch.distributed.launch --use_env --nproc_per_node=${GPUS_PER_NODE} --nnodes=${WORKER_CNT} --node_rank=${RANK} \ --master_addr=${MASTER_ADDR} --master_port=${MASTER_PORT} cn_clip/training/main.py \ --train-data=${train_data} \ --val-data=${val_data} \ --resume=${resume} \ ${reset_data_offset} \ ${reset_optimizer} \ --logs-specs=${output_base_dir} \ --name=${name} \ --save-step-frequency=${save_step_frequency} \ --save-epoch-frequency=${save_epoch_frequency} \ --log=${log_interval} \ ${report_training_batch_acc} \ --context-length=${context_length} \ --warmup=${warmup} \ --batch-size=${batch_size} \ --valid-batch-size=${valid_batch_size} \ --valid-step-interval=${valid_step_interval} \ --valid-epoch-interval=${valid_epoch_interval} \ --accum-freq=${accum_freq} \ --lr=${lr} \ --wd=${wd} \ --max-epochs=${max_epochs} \ --vision-model=${vision_model} \ ${use_augment} \ --text-model=${text_model} 中的 --log=${log_interval} 直接注释掉

不行啊 注释掉后报错：(clip-cn) ***@***.***:~/fsy/Chinese-CLIP$ bash run_scripts/muge_finetune_vit-b-16_rbt-base.sh /home/fanzhengjie/fsy/Clip-Data
/home/fanzhengjie/.conda/envs/clip-cn/lib/python3.10/site-packages/torch/distributed/launch.py:208: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use-env is set by default in torchrun.
If your script expects --local-rank argument to be set, please
change it to read from os.environ['LOCAL_RANK'] instead. See
https://pytorch.org/docs/stable/distributed.html#launch-utility for
further instructions

main()
usage: launch.py [-h] [--nnodes NNODES] [--nproc-per-node NPROC_PER_NODE] [--rdzv-backend RDZV_BACKEND] [--rdzv-endpoint RDZV_ENDPOINT] [--rdzv-id RDZV_ID]
[--rdzv-conf RDZV_CONF] [--standalone] [--max-restarts MAX_RESTARTS] [--monitor-interval MONITOR_INTERVAL]
[--start-method {spawn,fork,forkserver}] [--role ROLE] [-m] [--no-python] [--run-path] [--log-dir LOG_DIR] [-r REDIRECTS] [-t TEE]
[--local-ranks-filter LOCAL_RANKS_FILTER] [--node-rank NODE_RANK] [--master-addr MASTER_ADDR] [--master-port MASTER_PORT]
[--local-addr LOCAL_ADDR] [--logs-specs LOGS_SPECS] [--use-env]
training_script ...
launch.py: error: ambiguous option: --logs=/home/fanzhengjie/fsy/Clip-Data/experiments/ could match --logs-specs, --logs_specs

**Comment by camel2012** [comment]
在脚本里调cn_clip/training/main.py前，加一个 --\换行就好了，如下
          --master_addr=${MASTER_ADDR} --master_port=${MASTER_PORT} \
           -- \
	  cn_clip/training/main.py \
